components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.ts

```diff
@@ -162,57 +200,50 @@ export class ReportingComponent implements OnInit, OnDestroy {
       // ignore new term if same as previous term
       distinctUntilChanged(),
       // include currently selected report filters
-      withLatestFrom(this.reportQuery.filters),
+      withLatestFrom(this.reportQuery.state),
       // switch to new search observable each time the term changes
-      switchMap(([terms, filters]) => {
+      switchMap(([terms, reportQuery]) => {
         const { type, text } = terms;
         if (text && text.length > 0) {
-          return this.getSuggestions(type, text, filters);
+          return this.getSuggestions(type, text, reportQuery);
         }
         return observableOf([]);
       }),
       takeUntil(this.isDestroyed)
-    )
-      .subscribe(suggestions => this.availableFilterValues = suggestions.filter(e => e.text));
+    ).subscribe(suggestions => this.availableFilterValues = suggestions.filter(e => e.text));
+
+    this.filters$ = this.reportQuery.state.pipe(map((reportQuery: ReportQuery) =>
+      reportQuery.filters.map(filter => {
+        filter.value.id = filter.value.text;
+        if (['profile_id', 'node_id', 'control_id'].indexOf(filter.type.name) >= 0) {
```

Comment:
For the 'profile_id', 'node_id', and 'control_id', look to check if we have the names of these objects to display in the filter bar.